### ISSUE TITLE
Can't remove mod on old worlds

### ISSUE THREAD

**Issue by Waffled-II** [Frozeh/infinity-cave#8]
Attempting to load any worlds without the mod results in the world being unable to load, rather just saying "Failed to load datapacks, can't proceed with server load. You can either fix your datapacks or reset to vanilla with safemode."
For more context, I have a long-time survival world that doesn't play well with the mod so I needed to remove it, only to find out that I can't.
[latest.log](https://github.com/Frozeh/infinity-cave/files/14412768/latest.log)

**Comment by CharlieMan014** [comment]
Yeah I wanted to remove it because now I can't hear rain and it passes through all blocks just to figure out I can't even uninstall it despite not even loading any new chunks

**Comment by Frozeh** [comment]
Go into level.dat and remove it from there.

**Comment by illegalbugtrader** [comment]
Ive removed it from loadingmodlist and i still get the same error

**Comment by illegalbugtrader** [comment]
> Yeah I wanted to remove it because now I can't hear rain and it passes through all blocks just to figure out I can't even uninstall it despite not even loading any new chunks

just rebuild the roofs of the house and its fixed, same thing happened to me

**Comment by privatecolemine** [comment]
Hope this helps...

I downloaded a completely fresh forge server, and then booted it up, let it populate it's folders and files. And then I took everything but the level.dat (old and new) of the corrupted server's world folder, and then replaced the folders in the new world folder. Booted it up, and kept all my old progress, buildings, everything.

**Comment by gummyfrog** [comment]
Hi, for anyone who needs to uninstall this (to fix heightmaps / mob spawning), here's the simplest way.

1. Remove Infinity Cave and create a new world with the same seed and world generation options as the broken world
2. Copy all of the data from the broken world **except** level.dat and level.dat_old into the new world.
3. (Optional, for worlds that run on a server:) Edit level.dat with an NBT editor to remove data/Player.
4. Profit

I would not advise installing this mod until it changes world generation in a non-destructive way.

**Comment by Frozeh** [comment]
Here is the reason why this is in beta: Worldgen is currently marked as experimental by Mojang, there is only so much I can do as this project is actually possible to be hosted completely server-side, requiring nothing more than a resource pack on the server. Adding mods that are marked as "beta" should be explanatory.

**Comment by gummyfrog** [comment]
Yep, totally understandable. It would be worth prominently highlighting the known issues on Modrinth, people may not expect even a beta mod with incompatibilities to damage their vanilla worlds.

**Comment by Frozeh** [comment]
Thanks for the suggestion, I will highlight this issue on the page soon after the next update.
